tests/lib/rules/no-early-return.js

```diff
@@ -0,0 +1,302 @@
+/**
+ * @author *****your name*****
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const RuleTester = require('eslint').RuleTester
+const rule = require('../../../lib/rules/no-early-return')
+
+const tester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 2020,
+    sourceType: 'module'
+  }
+})
+
+tester.run('no-early-return', rule, {
+  valid: [
+    {
+      filename: 'test.vue',
+      code: `
+      <script setup>
+        const foo = ref()
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script setup>
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        setup () {
+          const foo = ref()
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        setup () {
+          const foo = ref()
+          return { foo }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        setup () {
+          const foo = ref()
+          function bar () {
+            return 1
+          }
+          return { foo }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data () {
+          return { foo: true }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        asyncData () {
+          return { foo: true }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        asyncData (payload) {
+          return { foo: true }
+        }
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        data: () => ({ foo: true })
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        asyncData: () => ({ foo: true })
+      }
+      </script>
+      `
+    },
+    {
+      filename: 'test.vue',
+      code: `
+      <script>
+      export default {
+        setup () {
+          const foo = computed(() => {
+            if (s) {
+              return
+            }
+          })
+        }
+      }
+      </script>
+      `
+    }
+  ],
```

Comment:
I found a false positive in the next test source.

```vue
<script>
export default {
  methods: {
    fn () {
      if (a) {
        return {}; // <- false positive
      }
    }
  },
  setup () {
    const foo = ref()
    return { foo }
  }
}
</script>
```

I think we need to check the scope of ReturnStatement correctly.
The implementation of the return-in-emits-validator rule may be helpful.
https://github.com/vuejs/eslint-plugin-vue/blob/master/lib/rules/return-in-emits-validator.js

Reply:
Or I think you can use code path analysis like `consistent-return` rule.
https://eslint.org/docs/developer-guide/code-path-analysis
https://github.com/eslint/eslint/blob/main/lib/rules/consistent-return.js